AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/SharedPrefHandler.java

```diff
@@ -322,10 +322,21 @@ private IJExpression extractResValue(SharedPrefHolder holder, Element method, IR
 		case STRING:
 			resourceGetMethodName = "getString";
 			break;
+		case ARRAY:
+			resourceGetMethodName = "getStringArray";
+			break;
 		default:
 			break;
 		}
-		return holder.getContextField().invoke("getResources").invoke(resourceGetMethodName).arg(idRef);
+
+		JInvocation resourceInvocation = holder.getContextField().invoke("getResources").invoke(resourceGetMethodName).arg(idRef);
+
+		if (IRClass.Res.ARRAY.equals(res)) {
+			JInvocation asList = getClasses().ARRAYS.staticInvoke("asList");
+			JInvocation newHashMap = JExpr._new(getClasses().HASH_SET.narrow(getClasses().STRING));
+			resourceInvocation = newHashMap.arg(asList.arg(resourceInvocation));
```

Comment:
We should handle the empty array here as well (I think it is possible to define an empty string array resource or not?)